### ISSUE TITLE
`credhub --version` returns a segmentation violation when running `credhub --version` right after download

### ISSUE THREAD

**Issue by ishustava** [cloudfoundry-incubator/credhub-cli#28]
CLi version: `v1.5.1`

I ran `brew upgrade credhub-cli`, then `credhub --version` to check that it installed correctly,

and got:
```
$ credhub --version
CLI Version: 1.5.1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1297fc6]

goroutine 1 [running]:
github.com/cloudfoundry-incubator/credhub-cli/credhub.(*CredHub).Request(0x0, 0x13a4125, 0x3, 0x13a680e, 0xc, 0xc4201107e0, 0x0, 0x0, 0x10, 0x1353a20, ...)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/credhub/request.go:20 +0x26
github.com/cloudfoundry-incubator/credhub-cli/credhub.(*CredHub).find(0x0, 0x13a483d, 0x5, 0x13a454b, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/credhub/find.go:54 +0x1a4
github.com/cloudfoundry-incubator/credhub-cli/credhub.(*CredHub).FindAllPaths(0x0, 0x13b5c26, 0x3f, 0x0, 0x0, 0x0)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/credhub/find.go:26 +0x8a
github.com/cloudfoundry-incubator/credhub-cli/commands.PrintVersion(0x1065acd, 0x132a800)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/commands/version.go:22 +0x17f
github.com/cloudfoundry-incubator/credhub-cli/commands.init.2.func1()
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/commands/version.go:39 +0x22
reflect.Value.call(0x1326140, 0x15508d8, 0x193, 0x13a434b, 0x4, 0x0, 0x0, 0x0, 0x1095eb1, 0x154f8c0, ...)
	/usr/local/go/src/reflect/value.go:434 +0x91f
reflect.Value.Call(0x1326140, 0x15508d8, 0x193, 0x0, 0x0, 0x0, 0xc420071788, 0xc4200efae0, 0x109e5c4)
	/usr/local/go/src/reflect/value.go:302 +0xa4
github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags.(*Option).call(0xc420114d00, 0x0, 0x193, 0x15288c0)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags/option.go:384 +0x3ff
github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags.(*Option).set(0xc420114d00, 0x0, 0x12da719, 0x1345b20)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags/option.go:224 +0x27e
github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags.(*Parser).parseOption(0xc42001a3c0, 0xc4200e5170, 0x7ffeefbff8ca, 0x7, 0xc420114d00, 0x12ea301, 0x0, 0x7, 0x13a3f23)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags/parser.go:509 +0x866
github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags.(*Parser).parseLong(0xc42001a3c0, 0xc4200e5170, 0x7ffeefbff8ca, 0x7, 0x0, 0x7ffeefbff8ca, 0x7)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags/parser.go:564 +0xbb
github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc42001a3c0, 0xc42000a350, 0x1, 0x1, 0x100f148, 0x10, 0x1357920, 0x1, 0xc420106f30)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags/parser.go:265 +0x366
github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags.(*Parser).Parse(0xc42001a3c0, 0xc420106f30, 0x2, 0xc42001a3c0, 0x0, 0xc420013b90)
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/vendor/github.com/jessevdk/go-flags/parser.go:186 +0x73
main.main()
	/tmp/build/8dc578f1/go/src/github.com/cloudfoundry-incubator/credhub-cli/main.go:25 +0xd0
```

**Comment by patrickhuber** [comment]
Verified on Windows 10 and Ubuntu LTS 16.04 as well. See closed issue #32

**Comment by patrickhuber** [comment]
This message is due to a null reference being passed to the Request method in request.go

```golang
func (ch *CredHub) Request(method string, pathStr string, query url.Values, body interface{}) (*http.Response, error) {
	return ch.request(ch.Auth, method, pathStr, query, body)
}
```

Setting the api endpoint via the ```credhub api``` command sets the value and the error goes away. 

I would expect some better error messaging around the api endpoint not being set instead of throwing a hard error message. Some parameter guarding should fix this.

**Comment by crawsible** [comment]
@ishustava @patrickhuber Yep, this is gross. Scheduling.

**Comment by crawsible** [comment]
@ishustava @patrickhuber This has been addressed in 1.5.3. Can you give it a check and close the issue if you feel it's been addressed?

**Comment by ishustava** [comment]
@crawsible LGTM. Thanks!
